mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1446,6 +1446,89 @@ struct ChainedReduction final : OpRewritePattern<vector::ReductionOp> {
   }
 };
 
+/// Replace:
+///   elementwise(a, b)
+/// with:
+///   sc_a = shape_cast(a)
+///   sc_b = shape_cast(b)
+///   res = elementwise(sc_a, sc_b)
+///   return shape_cast(res)
+/// for which `a` and `b` are vectors of rank > 1 and have unit leading and/or
+/// trailing dimension.
```

Comment:
nit: (Feel free to ignore) It's not immediately clear the shape_casts are to drop unit dims (until reading the example, or the name of the struct below).

Reply:
This is good feedback and I am happy to incorporate it before landing this, thanks Ben!